Added tests for dayOfYear

Spiralis opened the pull request:
dayOfYear is implemented in CS-Schyntax, as a reference.

bretcope commented:
Please _don't_ commit `tests.json` when submitting to this repo. Just the pre-compiled files. I'd prefer you do a force push on this branch to remove that file, or do another PR or something. I don't want that edit in revision history. It's for the purpose of maintaining canonical version numbers.

bretcope commented on tests/daysOfYear:
```diff
@@ -0,0 +1,168 @@
+e:     ^
+
+--
+f: dom(-367)
```
Should be `doy`

bretcope commented on tests/daysOfYear:
```diff
@@ -0,0 +1,168 @@
+n: 2014-01-07
+
+--
+f: dom(20..<10)
```
Should be `doy`

bretcope commented:
I commented on the couple of missed day of month references. I think this test suite could probably benefit from one or two tests which aren't focused on January or December. I don't think it will actually catch anything, but it would feel more comprehensive. If you fix up that stuff, and what I previously commented, I think that should do it for the tests.

Afterwards there are a few things left to do:
- Merge the C# implementation PR. I'll comment on that separately.
- Update the README documentation.
- Update History.md (I'll do this)
- Update the [language grammar definition](https://github.com/schyntax/schyntax/blob/master/grammar/schyntax.jison). Pretty easy, but grammars can be weird if you're not used to them, so let me know if you need help.
- Update the Go and JavaScript implementations. I can do this if you're not familiar with those languages.
- Create tagged releases for the implementations (I'll do this)

You can add the README and grammar changes as commits on this PR if you'd like.

Spiralis commented:
I'll work through these asap. 
The language grammar and Go is unchartered territory though.

bretcope commented:
Okay, I can take care of the Go implementation. I'll do the grammar changes this time, and hopefully the example will make sense for future changes.

Spiralis committed:
Added tests for dayOfYear

dayOfYear is implemented in CS-Schyntax, as a reference.

Spiralis commented:
:sunglasses: 

I have completed the following:
- [x] Updated tests for Schyntax as propsed
- [x] Updated README for Schyntax
- [x] Updated dayOfYear PR for CS-Schyntax as suggested
- [x] Added dayOfYear PR for JS-Schyntax. Tests are passing. Review pending.

AFAIK, that leaves this (which I understood that you will handle):
- [x] Go implementation
- [x] Update Grammar 
- [x] Update History

bretcope commented:
Everything has been completed now. `daysOfYear` is officially part of the language and all three official ports have been updated. Thanks for contributing, and thanks for your patience.

Spiralis commented:
:smile: Happy to contribute :+1:
